pinot-tools/src/main/java/org/apache/pinot/tools/streams/PinotSourceGenerator.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.tools.streams;
+
+import java.util.List;
+import java.util.Properties;
+import org.apache.pinot.spi.stream.RowWithKey;
+
+
+/**
+ * Represents one Pinot Real Time Data Source that can constantly generate data
+ * For example it can be pulling a batch from Kafka, or polling some data via HTTP GET
+ * The generator will be driven by PinotRealtimeSource to keep producing into some downstream sink
+ */
+public interface PinotSourceGenerator extends AutoCloseable {
```

Comment:
SourceGenerator sounds like you are generating a subclass of PinotRealtimeSource. `PinotSourceDataGenerator` may be a better name.

Reply:
Thanks. Changed